src/data/set/enumerate.lean

```diff
@@ -2,34 +2,40 @@
 Copyright (c) 2017 Johannes Hölzl. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johannes Hölzl
-
-Enumerate elements of a set with a select function.
 -/
 import data.set.lattice
 import tactic.wlog
+
+/-!
+# Set enumeration
+
+This file allows enumeration of sets given a choice function.
+-/
+
 noncomputable theory
 
-open function set
+open function
 
 namespace set
 
 section enumerate
 parameters {α : Type*} (sel : set α → option α)
 
+/-- Given a choice function `sel`, enumerates the elements of a set. -/
 def enumerate : set α → ℕ → option α
 | s 0       := sel s
 | s (n + 1) := do a ← sel s, enumerate (s \ {a}) n
 
-lemma enumerate_eq_none_of_sel {s : set α} (h : sel s = none) : ∀{n}, enumerate s n = none
+lemma enumerate_eq_none_of_sel {s : set α} (h : sel s = none) : ∀ {n}, enumerate s n = none
 | 0       := by simp [h, enumerate]; refl
 | (n + 1) := by simp [h, enumerate]; refl
 
-lemma enumerate_eq_none : ∀{s n₁ n₂}, enumerate s n₁ = none → n₁ ≤ n₂ → enumerate s n₂ = none
-| s 0       m := assume : sel s = none, by simp [enumerate_eq_none_of_sel, this]
-| s (n + 1) m := assume h hm,
+lemma enumerate_eq_none : ∀ {s n₁ n₂}, enumerate s n₁ = none → n₁ ≤ n₂ → enumerate s n₂ = none
+| s 0       m := λ h _, enumerate_eq_none_of_sel h
+| s (n + 1) m := λ h hm,
```

Comment:
Is there a reason why you changed all `assume`s to `λ`s? The `have`/`assume`/`show` style of proofs is not deprecated as far as I know, although it's not so popular in mathlib (anymore).

Reply:
Oh I thought it was kind of old fashioned. Should I revert?

Reply:
I do not really care either way, so do whatever you prefer :)